pkg/web_app/lib/likes.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:html';
+
+import 'package:intl/intl.dart';
+import 'package:mdc_web/mdc_web.dart';
+
+import 'src/account.dart';
+import 'src/page_data.dart';
+
+Future _done;
+
+void setupLikes() {
+  final likes = document.querySelector('#likes');
+  final toggleButton = document.querySelector('#thumb-up') as ButtonElement;
+
+  if (toggleButton == null) return;
+
+  final iconButtonToggle = MDCIconButtonToggle(toggleButton);
+  int likesDelta = 0;
+
+  String likesString() {
+    final likesCount = pageData.pkgData.likes + likesDelta;
+    return '${NumberFormat.compact().format(likesCount)}'
+        ' ${likesCount == 1 ? 'Like' : 'Likes'}';
+  }
+
+  iconButtonToggle.listen(MDCIconButtonToggle.changeEvent, (Event e) {
+    toggleButton.blur();
+    if (iconButtonToggle.on) {
+      // The button has shifted to on.
+      likesDelta++;
+      likes.innerHtml = likesString();
```

Comment:
Can we use `textContent` instead? If we're only updating text it's much better to use `textContent` as `innerHtml` could be leverage to do HTML injection (assuming we have another bug somewhere else)..